### DISCUSSION TITLE
periodical costs and constraints/CU are switched at the pareto point 1 and 0

### DISCUSSION THREAD

**Discussion by tschuering23** [SESMG/SESMG#215] [Q&A]
**Code of Conduct**
- [x] Hereby I consent to the [CODE_OF_CONDUCT](https://github.com/SESMG/SESMG/blob/master/CODE_OF_CONDUCT.md)[^1]  

**Describe the bug**
periodical costs und constraints/CU werden beim pareto-Punkt 1 und 0 vertauscht

**Expected behavior**
Die Kosten von 850 CU/kW sollen in der Spalte periodical costs stehen

**Screenshots**
![Screenshot 2023-12-08 132930](https://github.com/SESMG/SESMG/assets/153288975/78b054d2-8afd-47c4-8c69-c9a2949236a7)
![Screenshot 2023-12-08 133132](https://github.com/SESMG/SESMG/assets/153288975/48fdd95f-4e87-4fcc-aec2-be79a81fce70)
<img width="928" alt="Screenshot 2023-12-08 132719" src="https://github.com/SESMG/SESMG/assets/153288975/c0d7ae59-aebe-4e44-bfb4-691f2b9b8799">

**Desktop (please complete the following information):**
 - OS: [e.g. iOS 13.5.1]
 - Browser [mircosoft edge]
 - Python-version [3.9.13]

**Additional context**

[^1]: Fill in a x between the brackets to check the check box.

**Comment by chrklemm** [comment]
Hello @tschuering23 ! Thank you for supporting the development of the SESMG with your contribution. The connection described above is not a bug, but has been implemented by us for the following reason:

**"cost"** and **"constraint cost"** can refer to any linearly scalable variable. These can be financial costs, any kind of emissions or other environmental impacts. When optimizing with the SESMG, there is a primary optimization variable ("costs") which is optimized and a secondary optimization variable ("constraint costs") for which a limiting upper limit is defined.

In the epsilon-constraint method, which is used to determine the Pareto front, the assignment of the primary and secondary criteria varies:

**1st run (determination of the financial minimum)**: Financial costs = primary criterion ("costs"), ghg emissions = secondary criterion ("constraint costs")
**2nd run (determination of the financial minimum)**: Financial costs = secondary criterion ("constraint costs"), ghg emissions = primary criterion ("costs")
**All further runs (paret points)**: Financial costs = primary criterion ("costs"), ghg emissions = secondary criterion ("constraint costs")

In the 2nd run, the optimization criteria are therefore swapped with respect to the other runs and are also displayed the other way round in the results. This correlation is taken into account when displaying the Pareto front and the curve is displayed correctly.

I hope this helps.
Best regards,
@chrklemm
